lib/tld_sync.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Domain Manager TLD Sync Tool
+ *
+ * @link https://www.blesta.com Blesta
+ */
+class TldSync
+{
+    /**
+     * Synchronize the price for the given TLDs with those set by the registrar
+     *
+     * @param array $tlds A list of TLDs for which to sync registrar prices
+     * @param int $company_id The ID of the company where the TLDs will be synchronized (optional)
+     */
+    public function synchronizePrices(array $tlds, $company_id = null)
+    {
+        Loader::loadModels($this, ['Domains.DomainsTlds', 'ModuleManager']);
+
+        if (is_null($company_id)) {
+            $company_id = Configure::get('Blesta.company_id');
+        }
+
+        // Get TLD records
+        $tld_records = $this->DomainsTlds->getAll(
+            ['tlds' => $tlds, 'company_id' => $company_id]
+        );
+
+        // Organize TLD records by registrar module
+        $module_tlds = [];
+        foreach ($tld_records as $tld_record) {
+            $module_tlds[$tld_record->module_id][] = $tld_record->tld;
+        }
+
+        // Get TLD prices from the registrar module
+        foreach ($module_tlds as $module_id => $list_tlds) {
+            $module = $this->ModuleManager->initModule($module_id);
+            $module->setModuleRow($module->getModuleRows()[0] ?? null);
+            $module_pricing = $module->getTldPricing();
+
+            // Set the price for each TLD
+            $tlds_pricing = array_intersect_key($module_pricing, array_flip($list_tlds));
+            foreach ($tlds_pricing as $tld => $pricing) {
+                $this->DomainsTlds->updatePricings(
+                    $tld,
+                    $this->formatPricing($pricing, $tld, $company_id),
+                    $company_id
+                );
+            }
+        }
+    }
+
+    /**
+     * Format a pricing record returned by the registrar module
+     *
+     * @param array $pricing TLDs pricing
+     *    [currency => [year# => ['register' => price, 'transfer' => price, 'renew' => price]]]
+     * @param string $tld The TLD to sync registrar prices
+     * @param int $company_id The ID of the company where the TLDs will be synchronized (optional)
+     * @return array The formatted pricing record
+     */
+    private function formatPricing($pricing, $tld, $company_id = null)
+    {
+        Loader::loadModels($this, ['Domains.DomainsTlds']);
+
+        if (is_null($company_id)) {
+            $company_id = Configure::get('Blesta.company_id');
+        }
+
+        // Set TLD rounding
+        $tld_rounding = null;
+        if (($tld_settings['domains_enable_rounding'] ?? 0) == 1) {
+            $tld_rounding = $tld_settings['domains_markup_rounding'] ?? '.00';
+        }
+
+        // Fetch TLD sync settings
+        $tld_settings = $this->DomainsTlds->getDomainsCompanySettings();
```

Comment:
This needs to be moved above the evaluation on line 71 of $tld_settings['domains_enable_rounding']